### PR TITLE
validate manifest work executor permission on hub cluster

### PR DESCRIPTION
**Pull request by zhujian7**
Signed-off-by: zhujian <jiazhu@redhat.com>

Issue reference: https://github.com/open-cluster-management-io/work/issues/156

### CODE REVIEW

**Comment by zhujian7** [comment]
cc @qiujian16 @elgnay

**Comment by qiujian16** [comment]
it looks good in general, but it needs e2e test.

**Comment by qiujian16** [comment]
also how a user can set it when installing klusterlet?

**Comment by zhujian7** [comment]
> also how a user can set it when installing klusterlet?

The webhook will be deployed on the hub cluster, so this should be configured on the `ClusterManager`, right? I think we can use the [clustermanager.spec.registrationConfiguration.featureGates](https://github.com/open-cluster-management-io/api/blob/efae592d440845b83693239cedfefe4069508469/operator/v1/types.go#L59) to do that.

**Comment by zhujian7** [comment]
/hold
need to add e2e tests

File: pkg/webhook/manifestwork_webhook.go [hunk 1]
```diff
@@ -129,3 +152,57 @@ func (a *ManifestWorkAdmissionHook) validateManifest(manifest []byte) error {
...
+		}
+	}
+
+	if work.Spec.Executor == nil {
+		executor = &workv1.ManifestWorkExecutor{
+			Subject: workv1.ManifestWorkExecutorSubject{
+				Type: workv1.ExecutorSubjectTypeServiceAccount,
+				ServiceAccount: &workv1.ManifestWorkSubjectServiceAccount{
+					// give the default value "system:serviceaccount::klusterlet-work-sa"
+					Namespace: "",                   // TODO: Not sure what value is reasonable
```
Comment (zhujian7) [inline pkg/webhook/manifestwork_webhook.go:183]: we use empty here, which means hub users who want to use nil executor need `execute-as` permission on manifestwork with resource name `system:serviceaccount::klusterlet-work-sa`

**Comment by zhujian7** [comment]
/hold cancel
e2e tests are added
@qiujian16 ptal

**Comment by qiujian16** [comment]
/approve

need another review from @elgnay

**Pull request merged**